clang/test/dpct/helper_files_ref/include/dpl_extras/dpcpp_extensions.h

```diff
@@ -591,9 +595,31 @@ namespace device {
 namespace detail {
 
 template <typename... _Args> constexpr auto __joint_reduce(_Args... __args) {
+#ifdef SYCL_EXT_ONEAPI_USER_DEFINED_REDUCTIONS 
+  return sycl::ext::oneapi::experimental::joint_reduce(__args...);
+#else
   return sycl::joint_reduce(__args...);
+#endif
 }
 
+template <typename _Tp, typename... _Ts>
```

Comment:
1. Please don't make experimental API by default on. 2. put expermental API based code into experimental:: namespece.   Suggest introduce a new segmented_reduce function into experimental:: namespace and put all impment there.

Reply:
done